src/wp-includes/formatting.php

```diff
@@ -5847,36 +5847,32 @@ function wp_spaces_regexp() {
 }
 
 /**
- * Prints the important emoji-related styles.
+ * Enqueue the important emoji-related styles.
  *
- * @since 4.2.0
+ * @since 6.4.0
  */
-function print_emoji_styles() {
-	static $printed = false;
-
-	if ( $printed ) {
-		return;
+function enqueue_emoji_styles() {
+	$action = is_admin() ? 'admin_print_styles' : 'wp_print_styles';
+	if ( has_action( $action, 'print_emoji_styles' ) ) {
+		remove_action( $action, 'print_emoji_styles' );
```

Comment:
Two things:

Maybe this should rather short-circuit instead of putting the main logic inside. And secondly, I think this needs a good explanatory comment.

```php
/*
 * The following has_action() check is done for the sake of legacy code that would be 
 * used to disable the emoji, namely: 
 * 
 *     remove_action( 'wp_head', 'print_emoji_styles' );
 *
 * The only purpose for this action now is to be a legacy signal for whether to print
 * the emoji styles.
 */
$action = is_admin() ? 'admin_print_styles' : 'wp_print_styles';
if ( ! has_action( $action, 'print_emoji_styles' ) ) {
	return;
}
remove_action( $action, 'print_emoji_styles' );
```

See similar comment added for `wp_oembed_add_host_js()`:

https://github.com/WordPress/wordpress-develop/blob/414a1232c273f6e9dcf544d8ee411b22e5d92a5c/src/wp-includes/embed.php#L356-L397